test/unit/language-extensions/operators.spec.ts

```diff
@@ -0,0 +1,379 @@
+import * as path from "path";
+import * as util from "../../util";
+import * as tstl from "../../../src";
+import { invalidOperatorMappingUse } from "../../../src/transformation/utils/diagnostics";
+import { LuaTarget } from "../../../src";
+import { unsupportedForTarget } from "../../../src/transformation/utils/diagnostics";
+
+const operatorsProjectOptions: tstl.CompilerOptions = {
+    luaTarget: LuaTarget.Lua53,
+    types: [path.resolve(__dirname, "../../../language-extensions")],
+};
+
+const operatorMapTestObject = `
+    class OpTest {
+        public value: number;
+        public constructor(value: number) { this.value = value; }
+
+        public __add(other: OpTest): OpTest;
+        public __add(other: number): number;
+        public __add(other: OpTest | number) {
+            if (typeof other === "number") {
+                return this.value + other;
+            } else {
+                return new OpTest(this.value + other.value);
+            }
+        }
+        public __sub(other: OpTest) { return new OpTest(this.value - other.value); }
+        public __mul(other: OpTest) { return new OpTest(this.value * other.value); }
+        public __div(other: OpTest) { return new OpTest(this.value / other.value); }
+        public __mod(other: OpTest) { return new OpTest(this.value % other.value); }
+        public __pow(other: OpTest) { return new OpTest(this.value ** other.value); }
+        public __idiv(other: OpTest) { return new OpTest(Math.floor(this.value / other.value)); }
+        public __band(other: OpTest) { return new OpTest(this.value & other.value); }
+        public __bor(other: OpTest) { return new OpTest(this.value | other.value); }
+        public __bxor(other: OpTest) { return new OpTest(this.value ^ other.value); }
+        public __shl(other: OpTest) { return new OpTest(this.value << other.value); }
+        public __shr(other: OpTest) { return new OpTest(this.value >>> other.value); }
+
+        public __lt(other: OpTest) { return this.value < other.value; }
+        public __gt(other: OpTest) { return this.value > other.value; }
+
+        public __concat(other: OpTest) { return this.value.toString() + other.value.toString(); }
+
+        public __unm() { return -this.value; }
+        public __bnot() { return ~this.value; }
+        public __len() { return this.value; }
+    }
+`;
+
+const binaryMathOperatorTests: Array<[string, number, number, number]> = [
+    ["LuaAdd", 7, 4, 11],
```

Comment:
It's not obvious  what this means at first, might be more readable if you use objects instead of tuples, ie `{ extensionType: "LuaAdd", left: 7, right: 4, expectedResult: 11 }`, similar to https://github.com/TypeScriptToLua/TypeScriptToLua/blob/master/test/unit/assignments.spec.ts#L408